File: native/mobile-intelligence/android/BlackstarMobileBridge.kt
package blackstar.mobile.intelligence

/**
 * Native Android adapter boundary for Blackstar Mobile Intelligence.
 * Gemini Nano / ML Kit GenAI and AppFunctions availability must be capability-detected at runtime.
 */
data class BlackstarMobileCapabilityReport(
    val platform: String = "android",
    val osVersion: String,
    val nativeIntelligenceAvailable: Boolean,
    val nativeProvider: String?,
    val capabilities: List<String>,
    val appActionsAvailable: Boolean,
)

object BlackstarMobileBridge {
    fun report(
        osVersion: String,
        nativeIntelligenceAvailable: Boolean,
        capabilities: List<String>,
        appActionsAvailable: Boolean,
    ) = BlackstarMobileCapabilityReport(
        osVersion = osVersion,
        nativeIntelligenceAvailable = nativeIntelligenceAvailable,
        nativeProvider = if (nativeIntelligenceAvailable) "gemini-nano" else null,
        capabilities = capabilities.distinct().sorted(),
        appActionsAvailable = appActionsAvailable,
    )
}

File: native/mobile-intelligence/ios/BlackstarMobileBridge.swift
import Foundation

/// Native iOS adapter boundary for Blackstar Mobile Intelligence.
/// App Intents and Foundation Models availability must be detected by the app at runtime.
struct BlackstarMobileCapabilityReport: Codable {
    let platform: String = "ios"
    let osVersion: String
    let nativeIntelligenceAvailable: Bool
    let nativeProvider: String?
    let capabilities: [String]
    let appActionsAvailable: Bool
}

enum BlackstarMobileBridge {
    static func report(
        nativeIntelligenceAvailable: Bool,
        capabilities: [String],
        appActionsAvailable: Bool
    ) -> BlackstarMobileCapabilityReport {
        BlackstarMobileCapabilityReport(
            osVersion: ProcessInfo.processInfo.operatingSystemVersionString,
            nativeIntelligenceAvailable: nativeIntelligenceAvailable,
            nativeProvider: nativeIntelligenceAvailable ? "apple-foundation-models" : nil,
            capabilities: Array(Set(capabilities)).sorted(),
            appActionsAvailable: appActionsAvailable
        )
    }
}
